src/neptune/metadata_containers/utils.py

```diff
@@ -115,3 +131,30 @@ def prepare_nql_query(
 
     query = NQLQueryAggregate(items=query_items, aggregator=NQLAggregator.AND)
     return query
+
+
+def parse_dates(leaderboard_entries: Iterable[LeaderboardEntry]) -> Generator[LeaderboardEntry, None, None]:
+    yield from [_parse_entry(entry) for entry in leaderboard_entries]
+
+
+def _parse_entry(entry: LeaderboardEntry) -> LeaderboardEntry:
+    try:
+        return LeaderboardEntry(
+            entry.id,
+            attributes=[
+                AttributeWithProperties(
+                    attribute.path,
+                    attribute.type,
+                    {
+                        **attribute.properties,
+                        "value": datetime.strptime(attribute.properties["value"], DATE_FORMAT),
+                    },
+                )
+                if attribute.type == AttributeType.DATETIME
+                else attribute
+                for attribute in entry.attributes
+            ],
+        )
+    except (TypeError, ValueError):
+        # date is already in the right format or the parsing format is incorrect
+        return entry
```

Comment:
Looks good, I'd only add logging that date parsing fails instead of only defaulting to str value